test/suite/run.py

```diff
@@ -511,12 +529,31 @@ def error(exitval, prefix, msg):
     if debug:
         import pdb
         pdb.set_trace()
+    if batchtotal != 0:
+        # For test batching, we want to split up all the tests evenly, and
+        # spread out the tests, so each batch contains tests of all kinds. We'd
+        # like to prioritize the lowest scenario numbers first, so if there's a
+        # failure, we won't have to do all X thousand of some test's scenarios
+        # before we see a failure in the next test.  To that end, we define a
+        # sort function that sorts by scenario first, and test name second.
+        hugetests = set()
+        def batchsort(test):
```

Comment:
maybe we can rename "batchsort" to something like "get_sort_key". Initially, I thought this function will do the sorting, but it only extracts the pair of keys. My python knowledge is not the best, so ignore this comment if you are following a standard python nomenclature practice.

Reply:
Thanks, renamed it.